File: Yunhyunjo/Codingtest/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    map <string, int> com;
    map <string, int> par;
    for(int i = 0; i < completion.size(); i++){
        par[participant[i]]++;
        com[completion[i]]++;
    }
    par[participant[participant.size()-1]]++;
    
    for(int i = 0; i < participant.size(); i++){
        if(com[participant[i]] > 0){
            if(com[participant[i]] != par[participant[i]]){
                answer = participant[i];
                break;
            }
        }
        else{
            answer = participant[i];
            break;
        }
    }
    return answer;
}

File: gut27/Codingtest/완주하지못한선수.py
from collections import Counter
def solution(participant, completion):
    answer = ''
    
    participant += completion
    count = Counter(participant)

    if count.most_common()[-1][1] == 1:
        answer = count.most_common()[-1][0]
    else:
        for k,v in count.items():
            if v % 2 != 0:
                answer = k 
    return answer
